api/lib/ZwaveClient.ts

```diff
@@ -6946,6 +6942,97 @@ class ZwaveClient extends TypedEventEmitter<ZwaveClientEventCallbacks> {
 			}
 		}, 1000)
 	}
+
+	/**
+	 * Setup driver logging based on the configured format (text or JSON)
+	 * Uses the same logFormat setting as the gateway for consistency
+	 */
+	private setupDriverLogging(zwaveOptions: PartialZWaveOptions) {
+		// Get the log format from gateway settings - this applies to both app and driver logging
+		const settings = jsonStore.get(store.settings)
+		const logFormat = settings?.gateway?.logFormat || 'text'
+
+		if (logFormat === 'json') {
+			// JSON logging for driver - create custom transports for all outputs
+			const transports = []
+
+			// Custom console transport for JSON output
+			const jsonConsoleTransport = new winston.transports.Console({
+				format: winston.format.combine(
+					winston.format.timestamp(),
+					winston.format.json()
+				)
+			})
+			transports.push(jsonConsoleTransport)
+			
+			// Custom file transport for JSON output (if file logging is enabled)
+			// Use DailyRotateFile to maintain symlink functionality and proper date handling
+			if (this.cfg.logToFile) {
+				const fileTransport = new DailyRotateFile({
+					filename: ZWAVEJS_LOG_FILE,
+					auditFile: ZWAVEJS_LOG_FILE.replace('_%DATE%', '_logrotate').replace('.log', '.json'),
+					datePattern: 'YYYY-MM-DD',
+					createSymlink: true,
+					symlinkName: 'zwavejs_current.log',
+					zippedArchive: true,
+					maxFiles: `${this.cfg.maxFiles || 7}d`,
+					maxSize: '50m',
+					format: winston.format.combine(
+						winston.format.timestamp(),
+						winston.format.json()
+					)
+				})
+				transports.push(fileTransport)
+			}
+
+			// Custom WebSocket transport for JSON output
+			const jsonTransport = new JSONTransport()
+			jsonTransport.format = winston.format.combine(
+				winston.format.timestamp(),
+				winston.format.json()
+			)
+			transports.push(jsonTransport)
+
+			// Configure driver to use ONLY our custom transports
+			// Set logToFile: true to prevent the driver's internal console transport from being created
+			// Set forceConsole: false to ensure no internal console transport
+			// Redirect driver's internal file transport to /dev/null to prevent text logs
+			zwaveOptions.logConfig = {
+				...zwaveOptions.logConfig,
+				logToFile: true,       // This prevents internal console transport creation
+				forceConsole: false,   // This ensures no internal console transport
+				filename: '/dev/null', // Redirect driver's internal file transport to /dev/null
```

Comment:
This will break on Windows systems, which we do support.

Also, instead of hacking with the different file/console settings, simply set `enabled: false`, which will disable the internal transports.
You can still provide your own which will not be affected by the `enabled` flag.